src/battle_util.c

```diff
@@ -11250,11 +11283,18 @@ bool8 AreBattlersOfOppositeGender(u8 battler1, u8 battler2)
     return (gender1 != MON_GENDERLESS && gender2 != MON_GENDERLESS && gender1 != gender2);
 }
 
-u32 CalcSecondaryEffectChance(u8 battler, u8 secondaryEffectChance)
```

Comment:
In order to avoid the current workaround for Triple Arrows, you can change the function to
`CalcSecondaryEffectChance(u8 battler, u8 secondaryEffectChance, u16 moveEffect)`

That way for Triple Arrows, you could call this function once for the effect of lowering defense, and again for the effect of flinching.

Reply:
```
RandomPercentage(RNG_TRIPLE_ARROWS_DEFENSE_DOWN, CalcSecondaryEffectChance(gBattlerAttacker, 50, moveEffect));
RandomPercentage(RNG_TRIPLE_ARROWS_FLINCH, CalcSecondaryEffectChance(gBattlerAttacker, 30, EFFECT_FLINCH_HIT));

```
Just to make sure I understood you correctly. I made the suggested change and turned the Triple Arrow workaround into the previous 2 calls while adding `EFFECT_FLINCH_HIT` to the flinch check. Would this be correct?

Reply:
It would work, but for clarity pass in `EFFECT_DEFENSE_DOWN_HIT` explicitly instead of `moveEffect` on the first line

Reply:
Thx, it looks much cleaner now.